src/firmware_new/boards/frankie_v1/io/proto_multicast_communication.c

```diff
@@ -0,0 +1,242 @@
+        pb_encode(&stream, sender_profile->message_fields,
+                  sender_profile->protobuf_struct);
+        io_proto_multicast_communication_profile_releaseLock(sender_profile);
+
+        netbuf_alloc(tx_buf, stream.bytes_written);
+
+        // package payload and send over udp
+        tx_buf->p->payload = buffer;
+        netconn_sendto(conn, tx_buf, &sender_profile->multicast_address,
+                       sender_profile->port);
+    }
+
+    // we should never get here
+    netbuf_delete(tx_buf);
+    netconn_delete(conn);
+}
+
+void io_proto_multicast_listener_task(void* communication_profile)
+{
+    ProtoMulticastCommunicationProfile_t* listener_profile =
+        (ProtoMulticastCommunicationProfile_t*)communication_profile;
+
+    osEventFlagsWait(networking_event, NETIF_CONFIGURED, osFlagsWaitAny, osWaitForever);
+
+    // Bind the socket to the multicast address and port we then use that
+    // communication profile to join the specified multicast group.
+    struct netconn* conn = netconn_new(NETCONN_UDP_IPV6);
+    netconn_set_ipv6only(conn, true);
+    netconn_set_recvtimeout(conn, NETWORK_TIMEOUT_MS);
+
+    netconn_bind(conn, &listener_profile->multicast_address, listener_profile->port);
+    netconn_join_leave_group(conn, &listener_profile->multicast_address, IP6_ADDR_ANY,
+                             NETCONN_JOIN);
+
+    struct netbuf* rx_buf = NULL;
+    err_t network_err;
+    int no_protobuf_err;
+
+    for (;;)
+    {
+        network_err = netconn_recv(conn, &rx_buf);
+
+        switch (network_err)
+        {
+            case ERR_TIMEOUT:
+            {
+                io_proto_multicast_communication_profile_notifyEvents(listener_profile,
+                                                                      RECEIVE_TIMEOUT);
+                break;
+            }
+
+            case ERR_OK:
+            {
+                pb_istream_t in_stream = pb_istream_from_buffer(
+                    (uint8_t*)rx_buf->p->payload, rx_buf->p->tot_len);
+
+                // deserialize into buffer
+                io_proto_multicast_communication_profile_acquireLock(listener_profile);
+
+                // nanopb err logic is inverted, false = error
+                no_protobuf_err = pb_decode(&in_stream, listener_profile->message_fields,
+                                            listener_profile->protobuf_struct);
+
+                io_proto_multicast_communication_profile_releaseLock(listener_profile);
+
+                if (no_protobuf_err)
+                {
+                    io_proto_multicast_communication_profile_notifyEvents(
+                        listener_profile, RECEIVED_PROTO);
+                }
+                break;
+            }
+            default:
+            {
+                netbuf_delete(rx_buf);
+            }
+        }
+    }
+
+    // we should never get here
+    netconn_delete(conn);
+}
+
+void io_proto_multicast_startNetworkingTask(void* arg)
```

Comment:
Could we change all instances of `arg` to something a bit more descriptive?

Reply:
fixed